src/base/bittorrent/sessionimpl.cpp

```diff
@@ -4647,6 +4649,9 @@ void SessionImpl::handleTorrentResumeDataReady(TorrentImpl *const torrent, const
         m_resumeDataStorage->remove(iter.value());
         m_changedTorrentIDs.erase(iter);
     }
+
+    if (m_numResumeData == 0)
+        m_resumeDataStorage->makeBackup();
```

Comment:
Maybe only make backups at program starts?
If the db is big, this might be too slow.

Reply:
The application can run for quite a long time from the moment of start until the storage is damaged (which is more likely to occur when shutting down). If you make a copy only at the start, then backup copy may be too outdated. In addition, writing several tens of megabytes from time to time is not so much compared to the torrent in itself.
In any case, we need to start somewhere and wait for feedback.

Reply:
Here is a not uncommon usage: assuming a user has 10 MB database. Now he adds 10 .torrent files slowly, one by one. How many disk write for backup file occurred? 100 MB! And maybe the torrents were added in paused state and hasn't even start downloading. This is just too much IMO.

>If you make a copy only at the start, then backup copy may be too outdated. In addition, writing several tens of megabytes from time to time is not so much compared to the torrent in itself.

A better idea is to use "Save resume data interval" and generate backups on this period. But users might still argue that when all torrents are paused (no statistics were updated) it is wasteful to make such backups.
So maybe add another option (under advanced tab) for it? so users can tweak it.

Also is database corruption really a common thing? So far only one user reported it and the cause is his own fault (force killing qbt). Do we really need such feature?

Reply:
> Now he adds 10 .torrent files slowly, one by one. How many disk write for backup file occurred?

None. Resume data is initially saved in other place of code.



> A better idea is to use "Save resume data interval" and generate backups on this period. But users might still argue that when all torrents are paused (no statistics were updated) it is wasteful to make such backups.

If resume data isn't changed the saving is not requested.


> So maybe add another option (under advanced tab) for it? so users can tweak it.

This is certainly too much for the initial implementation, IMO.
>Also is database corruption really a common thing? 

Fortunately, it doesn't seem to be. But it does not become less dangerous to leave it without the slightest protection. Since in case of SQLite we have a single file, the user may lose all data if it is corrupted.

Reply:
> > Now he adds 10 .torrent files slowly, one by one. How many disk write for backup file occurred?
> 
> None. Resume data is initially saved in other place of code.
> 
> > A better idea is to use "Save resume data interval" and generate backups on this period. But users might still argue that when all torrents are paused (no statistics were updated) it is wasteful to make such backups.
> 
> If resume data isn't changed the saving is not requested.

Have you really see how frequent the backup file is written? The backup file is generated at torrent pause/resume/added, which hardly counts as from time to time, more like always update it.
IIRC "Save resume data interval" was tweaked to current 1 hour in order to minimize disk wearing.

@qbittorrent/bug-handlers 
Please test the disk wearing scenario mentioned in https://github.com/qbittorrent/qBittorrent/pull/17929#discussion_r1006420172. Notice that backup file modification time will be updated at every torrent add/pause/resume. Is this frequent disk updating acceptable if someone as some moderate size db file (>= 10MB)?

Reply:
I am open to any suggestions for tweaking it. I still think that doing a backup once when launching the application is not enough (of course, it's still better than not doing it at all). Alternatively, we can simply bind to "Save resume data interval".
>The backup file is generated at torrent add/pause/resume,

The counter question is how often do you do the above?

In any case, some fixed interval would be a good compromise, wouldn't it? But I don't want to add a separate option for it yet.

Reply:
>we can simply bind to "Save resume data interval".

OK. But the implementation should be smart enough to not make backups when the db has not changed from the last save.

Reply:
> But the implementation should be smart enough to not make backups when the db has not changed from the last save.

Sure.